src/main.js

```diff
@@ -730,10 +730,14 @@ class MiniGraphCard extends LitElement {
   updateBounds({ config } = this) {
     this.bound = [
       config.lower_bound !== undefined
-        ? config.lower_bound
+        ? !config.soft_bound
+          ? config.lower_bound
+          : Math.min(config.lower_bound, ...this.primaryYaxisSeries.map(ele => ele.min))
         : Math.min(...this.primaryYaxisSeries.map(ele => ele.min)) || this.bound[0],
       config.upper_bound !== undefined
-        ? config.upper_bound
+        ? !config.soft_bound
+          ? config.upper_bound
+          : Math.max(config.upper_bound, ...this.primaryYaxisSeries.map(ele => ele.max))
```

Comment:
Few things:
* double inline conditions are evil ;), please refactor
* you have added this option for main axis please do the same for secondary (to have consistency), I think you don't need a separate setting for it and you can reuse this one
* please update documentation

If I understand correctly you want to have a fixed bound but in case of value out of range we should render it correctly, righ?

@kalkih this seems to me like a bugfix - not a new feature, no? I'm trying to think about scenario where such hard bounds are useful but TBH nothing reasonable comes to my mind

Reply:
I think it could potentially be confusing for the user if the bound is specifically set but then also is dynamic, so I like the idea of having this as an option.
While I too can't find a use case where this isn't preferred, I'm sure there are, so we should keep that option available.

If it's documented well we could probably make this enabled by default though.